src/components/ImageView/index.js

```diff
@@ -57,12 +61,79 @@ class ImageView extends PureComponent {
         document.removeEventListener('mousemove', this.trackMovement.bind(this));
     }
 
+    onContainerLayoutChanged(e) {
+        const {width, height} = e.nativeEvent.layout;
+        const imageWidth = this.state.imgWidth;
+        const imageHeight = this.state.imgHeight;
+        const scale = imageHeight && imageWidth ? Math.min(width / imageWidth, height / imageHeight) : 0;
+        this.setState({
+            containerHeight: height,
+            containerWidth: width,
+            zoomScale: scale,
+        });
+    }
+
+    onContainerPressIn(e) {
+        const {pageX, pageY} = e.nativeEvent;
+        this.setState({
+            isMouseDown: true,
+            initialX: pageX,
+            initialY: pageY,
+            initialScrollLeft: this.scrollableRef.scrollLeft,
+            initialScrollTop: this.scrollableRef.scrollTop,
+        });
+    }
+
+    onContainerPress(e) {
+        if (this.state.isZoomed && !this.state.isDragging) {
+            const {offsetX, offsetY} = e.nativeEvent;
+            const delta = this.getScrollOffset(offsetX, offsetY);
+            const sX = delta.offsetX;
+            const sY = delta.offsetY;
+            this.scrollableRef.scrollTop = sY * this.state.zoomScale;
+            this.scrollableRef.scrollLeft = sX * this.state.zoomScale;
+        }
+
+        if (this.state.isZoomed && this.state.isDragging && this.state.isMouseDown) {
+            this.setState({isDragging: false, isMouseDown: false});
+        }
+    }
+
+    onContainerPressOut() {
+        if (this.state.isDragging) {
+            return;
+        }
+
+        this.setState(prevState => ({
+            isZoomed: !prevState.isZoomed,
+            isMouseDown: false,
+        }));
+    }
+
+
     /**
      * When open image, set image left/right/top/bottom point and width, height
-     * @param {Boolean} width image width
-     * @param {Boolean} height image height
+     * @param {Number} imageWidth
+     * @param {Number} imageHeight
      */
-    setImageRegion(width, height) {
+    setImageRegion(imageWidth, imageHeight) {
+        let width = imageWidth;
+        let height = imageHeight;
+        const containerHeight = this.state.containerHeight;
+        const containerWidth = this.state.containerWidth;
+
+
+        // return if image not loaded yet
+        if (imageHeight <= 0 || containerHeight <= 0) {
+            return;
+        }
+
+        // Fit the image to container size if image small than container.
+        const aspectRatio = Math.min((containerHeight / imageHeight), (containerWidth / imageWidth));
```

Comment:
```suggestion
        const aspectRatio = Math.min(containerHeight / imageHeight, containerWidth / imageWidth);
```

Remove unnecessary parenthesis